src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -349,22 +378,36 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             }
         }
 
-        private int FindEntry(TKey key)
+        private ref Entry FindEntry(TKey key, out bool found)
```

Comment:
I agree with @mikendn - this change should be evaluated separately from the Comparer change (submitting it as separate PR would be best).